bigbone/src/main/kotlin/social/bigbone/api/method/StatusMethods.kt

```diff
@@ -421,4 +423,108 @@ class StatusMethods(private val client: MastodonClient) {
             method = MastodonClient.Method.POST
         )
     }
+
+    /**
+     * Edit a given status to change its text, sensitivity, media attachments. Use [getStatus] and/or [getStatusSource]
+     * to get previous status data to use in this update.
+     * @param statusId the ID of the Status in the database
+     * @param status the plain text content of the status
+     * @param mediaIds the array of media ids to attach to the status (maximum 4)
+     * @param sensitive whether the status should be marked as sensitive
+     * @param spoilerText the plain text subject or content warning of the status
+     * @param language ISO 639 language code for the status
+     * @see <a href="https://docs.joinmastodon.org/methods/statuses/#edit">Mastodon API documentation: methods/statuses/#edit</a>
+     */
+    @JvmOverloads
+    @Throws(BigBoneRequestException::class)
+    fun editStatus(
+        statusId: String,
+        status: String,
+        mediaIds: List<String>? = null,
+        sensitive: Boolean = false,
+        spoilerText: String? = null,
+        language: String? = null
+    ): MastodonRequest<Status> {
+        return client.getMastodonRequest(
+            endpoint = "api/v1/statuses/$statusId",
+            method = MastodonClient.Method.PUT,
+            parameters = Parameters().apply {
+                append("status", status)
+                mediaIds?.let { append("media_ids", it) }
+                append("sensitive", sensitive)
+                spoilerText?.let { append("spoiler_text", it) }
+                language?.let { append("language", it) }
+            }
+        )
+    }
+
+    /**
+     * Edit a given status to change its text, sensitivity, or poll. Note that editing a poll’s options will reset the votes.
```

Comment:
Typo: Should be "Edit a given poll..."

Reply:
As discussed externally, the distinction between "polls" and "statuses" is one of our library - Mastodon considers both of them to be statuses. As such, it is possible to edit a status with poll using `editStatus()` if we don't want to touch its poll data, or potentially to edit a status _without_ poll using `editPoll()` to add one.